### ISSUE TITLE
Add cross-platform support for FileBrowser

### ISSUE THREAD

**Issue by rderooy** [joncampbell123/dosbox-x#1875]
**Is your feature request related to a problem? Please describe.**
Right now you can mount image files or directories in DOSBox-X from the menu only on Windows.

The purpose of this Feature Request is to make it cross-platform. Unfortunately for some reason SDL lacks native support for this. For this purpose I have been looking around for a cross-platform file browser solution, ideally using the native OS file dialogs. While also avoiding to link to huge external libraries like QT and wxWidget.

Here are some interesting options I have found, in order of preference:
https://github.com/mlabbe/nativefiledialog
https://github.com/AndrewBelt/osdialog
https://sourceforge.net/projects/tinyfiledialogs/

**Comment by Wengier** [comment]
I have checked all of them and think the last one is probably the best one for our purpose. The first one works fine for Windows but for Linux it requires external libraries (GTK or Zenity). The second one does not seem to work very well.

**Comment by Wengier** [comment]
So I have already enabled almost all mounting options (with the only exception of "Mount Automatically" which is apparently specific to Windows) including "Mount as Hard Disk", "Mount as CD-ROM", "Mount as Floppy", "Mount disk image" and "Boot from disk image" for all platforms in commit https://github.com/joncampbell123/dosbox-x/commit/291a31218a60e631c1d7a55e508659d8d06e6f43. Tested in both Windows and Linux. This is hopefully very useful for non-Windows users who previously did not have access to these mounting options.
